src/config.py

```diff
@@ -221,11 +246,12 @@ def server_properties(self) -> List[str]:
                 f"auto.create.topics={self.charm.config['auto-create-topics']}",
                 f"super.users={self.super_users}",
                 f"log.dirs={self.log_dirs}",
-                f"listeners={protocol}://:{port}",
-                f"advertised.listeners={protocol}://{host}:{port}",
-                f'listener.name.{(protocol).lower()}.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="sync" password="{self.sync_password}";',
-                f"security.inter.broker.protocol={protocol}",
-                f"security.protocol={protocol}",
+                f"listener.security.protocol.map=INTERNAL:{protocol},EXTERNAL:{protocol}",
+                f"listeners=INTERNAL://:{self.internal_port},EXTERNAL://:{self.external_port}",
+                f"advertised.listeners=INTERNAL://{host}:{self.internal_port},EXTERNAL://{self.external_listener}:{self.external_port}",
+                "inter.broker.listener.name=INTERNAL",
+                f'listener.name.internal.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="sync" password="{self.sync_password}";',
+                f'listener.name.external.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username="sync" password="{self.sync_password}";',
```

Comment:
Out of scope but noting here, my ticket will update this to an `admin` super-user instead of using the same user as the inter-broker `sync` user.